README.md

```diff
@@ -1,2 +1,79 @@
-# beacon-starter
+# Beacon Reader Example
 
+> A starter project for reading beacon values from a smart contract
+
+This project is composed of the following:
+
+1. A test that uses a mocked RrpBeaconServer to simulate reading a beacon value from a smart contract.
+1. A script to deploy the BeaconReaderExample contract to any network. This script will set the address of the RrpBeaconServer contract deployed on the selected network and after the script finishes it will display the address of the contract deployed but also save this address to `deployments` directory on the root of this repo. If selected network is not `hardhat` or `localhost` then this address must be then whitelisted by the RrpBeaconServer manager account prior to reading the beacon value.
+
+## Instructions
+
+1. Clone this repo
+1. Run the following to install the dependencies
+
+```sh
+npm install
+```
+
+Now depending on the network you are using you will need to do the following:
+
+### Default network
+
+Simplest way to test this project is by using the integrated [Hardhat network](https://hardhat.org/hardhat-network/) and not extra steps are needed in this case to run both the test and the deployment script.
+
+#### Test
+
+```sh
+hardhat test test/BeaconReaderExample.test.js
+```
+
+#### Script
+
+```sh
+hardhat run scripts/deploy.js
+hardhat run scripts/read-beacon.js
```

Comment:
The `deploy` step worked, but then ran into this error with  the `read-beacon` step:
```sh
$ npx hardhat run scripts/read-beacon.js
Error: call revert exception (method="readBeacon(bytes32)", errorArgs=null, errorName=null, errorSignature=null, reason=null, code=CALL_EXCEPTION, version=abi/5.5.0)
    at Logger.makeError (/home/derek/api3/beacon-reader-example/node_modules/@ethersproject/logger/src.ts/index.ts:225:28)
    at Logger.throwError (/home/derek/api3/beacon-reader-example/node_modules/@ethersproject/logger/src.ts/index.ts:237:20)
    at Interface.decodeFunctionResult (/home/derek/api3/beacon-reader-example/node_modules/@ethersproject/abi/src.ts/interface.ts:425:23)
    at Contract.<anonymous> (/home/derek/api3/beacon-reader-example/node_modules/@ethersproject/contracts/src.ts/index.ts:392:44)
    at step (/home/derek/api3/beacon-reader-example/node_modules/@ethersproject/contracts/lib/index.js:48:23)
    at Object.next (/home/derek/api3/beacon-reader-example/node_modules/@ethersproject/contracts/lib/index.js:29:53)
    at fulfilled (/home/derek/api3/beacon-reader-example/node_modules/@ethersproject/contracts/lib/index.js:20:58) {
  reason: null,
  code: 'CALL_EXCEPTION',
  method: 'readBeacon(bytes32)',
  errorArgs: null,
  errorName: null,
  errorSignature: null,
  address: '0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512',
  args: [
    '0xf797ab0ee716cbfaa29a1e7866ea69216d077c4aca6ce49b096b110e7ac76ce9'
  ],
  transaction: {
    data: '0x1de7e45bf797ab0ee716cbfaa29a1e7866ea69216d077c4aca6ce49b096b110e7ac76ce9',
    to: '0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512',
    from: '0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266',
    gasLimit: BigNumber { value: "29023448" }
  }
}
```